public/pages/DetectorResults/containers/AnomalyHistory.tsx

```diff
@@ -99,58 +118,70 @@ export const AnomalyHistory = (props: AnomalyHistoryProps) => {
   const isHCDetector = !isEmpty(detectorCategoryField);
   const backgroundColor = darkModeEnabled() ? '#29017' : '#F7F7F7';
 
-  useEffect(() => {
-    // We load at most 10k AD result data points for one call. If user choose
-    // a big time range which may have more than 10k AD results, will use bucket
-    // aggregation to load data points in whole time range with larger interval.
-    async function getBucketizedAnomalyResults() {
-      try {
-        setIsLoadingAnomalyResults(true);
-        const anomalySummaryResult = await dispatch(
-          searchResults(
-            getAnomalySummaryQuery(
-              dateRange.startDate,
-              dateRange.endDate,
-              props.detector.id
-            )
+  // We load at most 10k AD result data points for one call. If user choose
+  // a big time range which may have more than 10k AD results, will use bucket
+  // aggregation to load data points in whole time range with larger interval.
+  // If entity is specified, we only query AD result data points for this entity.
+  async function getBucketizedAnomalyResults(
+    entity: Entity | undefined = undefined
+  ) {
+    try {
+      setIsLoadingAnomalyResults(true);
+      const anomalySummaryResult = await dispatch(
+        searchResults(
+          getAnomalySummaryQuery(
+            dateRange.startDate,
+            dateRange.endDate,
+            props.detector.id,
+            entity
           )
-        );
+        )
+      );
+
+      setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
+      setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
 
-        setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
-        setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
-
-        const result = await dispatch(
-          searchResults(
-            getBucketizedAnomalyResultsQuery(
-              dateRange.startDate,
-              dateRange.endDate,
-              1,
-              props.detector.id
-            )
+      const result = await dispatch(
+        searchResults(
+          getBucketizedAnomalyResultsQuery(
+            dateRange.startDate,
+            dateRange.endDate,
+            1,
+            props.detector.id,
+            entity
           )
-        );
+        )
+      );
 
-        setBucketizedAnomalyResults(parseBucketizedAnomalyResults(result));
-      } catch (err) {
-        console.error(
-          `Failed to get anomaly results for ${props.detector.id}`,
-          err
-        );
-      } finally {
-        setIsLoadingAnomalyResults(false);
-      }
+      setBucketizedAnomalyResults(parseBucketizedAnomalyResults(result));
+    } catch (err) {
+      console.error(
+        `Failed to get anomaly results for ${props.detector.id}`,
+        err
+      );
+    } finally {
+      setIsLoadingAnomalyResults(false);
     }
+  }
 
+  useEffect(() => {
     fetchRawAnomalyResults(isHCDetector);
```

Comment:
From line 208, the parameter's name is actually showLoading.  Why is isHCDetector related to showLoading?

Reply:
If detector is HC detector, loading raw data is necessary for rendering anomaly result table like [anomalyResult](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/public/pages/DetectorResults/containers/AnomalyHistory.tsx#L408) which is from `atomicAnomalyResults`. Thus, in that case we need to have `showLoading` to be true so that UI can have loading status. 

In case of single entity detector, loading raw data is not necessary for rendering any UI result, because it is only used for feature missing checking. Thus, we don't need to set `showLoading` to be true in that case.